scm/pipeline.go

```diff
@@ -0,0 +1,106 @@
+package scm
+
+type (
+	ObjectAttributes struct {
+		ID         int
+		IID        int
```

Comment:
a lot of this data, such as `iid` is specific to harness and cannot be abstracted across providers